compose_api.go

```diff
@@ -0,0 +1,291 @@
+package testcontainers
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+
+	"github.com/compose-spec/compose-go/cli"
+	"github.com/compose-spec/compose-go/types"
+	"github.com/docker/compose/v2/pkg/api"
+	types2 "github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+type stackUpOptionFunc func(s *stackUpOptions)
+
+func (f stackUpOptionFunc) applyToStackUp(o *stackUpOptions) {
+	f(o)
+}
+
+type stackDownOptionFunc func(do *api.DownOptions)
+
+func (f stackDownOptionFunc) applyToStackDown(do *api.DownOptions) {
+	f(do)
+}
+
+// RunServices is comparable to 'docker-compose run' as it only creates a subset of containers
+// instead of all services defined by the project
+func RunServices(serviceNames ...string) StackUpOption {
+	return stackUpOptionFunc(func(o *stackUpOptions) {
+		o.Services = serviceNames
+	})
+}
+
+// IgnoreOrphans - Ignore legacy containers for services that are not defined in the project
+type IgnoreOrphans bool
+
+func (io IgnoreOrphans) applyToStackUp(co *api.CreateOptions, _ *api.StartOptions) {
+	co.IgnoreOrphans = bool(io)
+}
+
+// RemoveOrphans will cleanup containers that are not declared on the compose model but own the same labels
+type RemoveOrphans bool
+
+func (ro RemoveOrphans) applyToStackUp(o *stackUpOptions) {
+	o.RemoveOrphans = bool(ro)
+}
+
+func (ro RemoveOrphans) applyToStackDown(o *stackDownOptions) {
+	o.RemoveOrphans = bool(ro)
+}
+
+// Wait won't return until containers reached the running|healthy state
+type Wait bool
+
+func (w Wait) applyToStackUp(o *stackUpOptions) {
+	o.Wait = bool(w)
+}
+
+// RemoveImages used by services
+type RemoveImages uint8
+
+func (ri RemoveImages) applyToStackDown(o *stackDownOptions) {
+	switch ri {
+	case RemoveImagesAll:
+		o.Images = "all"
+	case RemoveImagesLocal:
+		o.Images = "local"
+	}
+}
+
+type ComposeStackFiles []string
+
+func (f ComposeStackFiles) applyToComposeStack(o *composeStackOptions) {
+	o.Paths = f
+}
+
+type StackIdentifier string
+
+func (f StackIdentifier) applyToComposeStack(o *composeStackOptions) {
+	o.Identifier = string(f)
+}
+
+func (f StackIdentifier) String() string {
+	return string(f)
+}
+
+const (
+	// RemoveImagesAll - remove all images used by the stack
+	RemoveImagesAll RemoveImages = iota
+	// RemoveImagesLocal - remove only images that don't have a tag
+	RemoveImagesLocal
+)
+
+type dockerComposeAPI struct {
+	lock           sync.RWMutex
+	name           string
+	configs        []string
+	waitStrategies map[string]wait.Strategy
+	containers     map[string]*DockerContainer
+	composeService api.Service
+	dockerClient   client.APIClient
+	projectOptions []cli.ProjectOptionsFn
+	project        *types.Project
+}
```

Comment:
It would be great if we add comments for the attributes, so it's usage is even simpler